### PR TITLE
Added unit tests for OcActionDrop

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Added following uni tests:
- toggle dropdown
    - when toggle button is toggled
    - when dropdown is open and  menu is clicked
- open menu and focus

## Related Issue

- Part of #1366 

## How Has This Been Tested?

- 🤖 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

## Open tasks:

- [ ] ...

### CODE REVIEW

File: src/components/OcActionDrop.vue
Comment (kiranparajuli589) [inline src/components/OcActionDrop.vue:79]: default is not a valid prop-value for oc-button variation prop

Comment (individual-it) [reply src/components/OcActionDrop.vue:79]: how can we add tests, so that it does not happen again?

Comment (kiranparajuli589) [reply src/components/OcActionDrop.vue:79]: validator used here and in the `OcButton` component will throw `vue [warn]` if the invalid prop is used.
we've to catch the console warning and test if waring for invalid props are not present, like in https://github.com/owncloud/owncloud-design-system/pull/1364/files where invalid prop vue warns are thrown as real error -> after merge we could use that before every test cases

**Review by individual-it — COMMENTED** [review]
LGTM, but would like to have a review from @kulmann @LukasHirt or @pascalwengerter because of the changes in the code

File: src/components/OcActionDrop.vue
Comment (kulmann) [reply src/components/OcActionDrop.vue:79]: Could you also add all other valid button variations? can be found here https://github.com/owncloud/owncloud-design-system/blob/64d5456127a30a6db5744cae9d0718fb1875476d/src/components/OcButton.vue#L93

File: src/components/OcActionDrop.spec.js
Comment (kulmann) [inline src/components/OcActionDrop.spec.js:32]:
```suggestion
    describe("when dropdown is open and menu is clicked", () => {
```

**Review by kulmann — CHANGES_REQUESTED** [review]

File: src/components/OcActionDrop.vue
Comment (kiranparajuli589) [reply src/components/OcActionDrop.vue:79]: fixed

**Review by saw-jan — COMMENTED** [review]
LGTM

File: src/components/OcActionDrop.spec.js
Comment (individual-it) [inline src/components/OcActionDrop.spec.js:45]: maybe check first if the dropdown is open, to make sure we don't have a snake oil test here

**Review by individual-it — CHANGES_REQUESTED** [review]
should we put `.oc-dropdown-menu` `.oc-dropdown-wrapper button` into a const?

File: src/components/OcActionDrop.spec.js
Comment (kiranparajuli589) [reply src/components/OcActionDrop.spec.js]: fixed

**Review by individual-it — APPROVED** [review]

**Pull request merged**
